NameLayerMain/src/vg/civcraft/mc/namelayer/group/Group.java

```diff
@@ -317,23 +318,38 @@ public PlayerType getCurrentRank(UUID uuid) {
 	 * @param type- The PlayerType to add. If a preexisting PlayerType is found, 
 	 * it will be overwritten.
 	 */
-	public void addMember(UUID uuid, PlayerType type) {
+	
+	public void addMember(UUID uuid, PlayerType type){
+		addMember(uuid,type,true);
+	}
+	
+	public void addMember(UUID uuid, PlayerType type, boolean savetodb) {
 		if (type == PlayerType.NOT_BLACKLISTED) {
 			return;
 		}
-		if (isMember(uuid, type)) {
-			db.removeMember(uuid, name);
+		if (savetodb == true) {
+			if (isMember(uuid, type)){
+				db.removeMemberAsync(uuid, name);
+			}
+			db.addMemberAsync(uuid, name, type);
+			Mercury.message("addMember " + this.name + " " + uuid.toString() + " " + type.toString());
 		}
 		players.put(uuid, type);
-		db.addMember(uuid, name, type);
 	}
 
 	/**
 	 * Removes the Player from the Group.
 	 * @param uuid- The UUID of the Player.
 	 */
-	public void removeMember(UUID uuid) {
-		db.removeMember(uuid, name);
+	public void removeMember(UUID uuid){
```

Comment:
Same concern here, and shared with all down the line.

Changing previously sync to async on previously published interface.